NeoML/include/NeoML/FineObjLite/ObjectFOL.h

```diff
@@ -462,11 +462,10 @@ inline const CCopyOnWritePtr<T>& CCopyOnWritePtr<T>::assignPtr( T* newPtr )
 	if( newPtr != 0 ) {
 		newPtr->addRef();
 	}
-	T* oldPtr = ptr;
```

Comment:
It isn't my invension, but I know the reason of a tmp variable here:
`Assigning a new value must be done before calling release metod on the old pointer so that in case of an exception when the old object is destroyed, CCopyOnWritePtr object remains in the correct state`
Maybe there should be a comment about it here.

Reply:
Exception in case of old object destroying it's-direct way to terminate program, and no matter about correct state.